.github/workflows/smoke-test.yaml

```diff
@@ -56,8 +71,10 @@ jobs:
           make -C install/kubernetes docs
           git diff --exit-code || (echo "please run 'make -C install/kubernetes docs' and submit your changes"; exit 1)
 
+  # Doesn't run if the only modified files were under test/ or Documentation/.
   conformance-test:
-    if: ${{ github.repository == 'cilium/cilium' }}
+    needs: check_changes
+    if: ${{ github.repository == 'cilium/cilium' && needs.check_changes.outputs.tested == 'true' }}
```

Comment:
For reviewers: I did not extend this condition to other jobs in the same file because I believe they are very short and it's probably not worth it. Happy to change that though.